src/main/java/org/alliancegenome/curation_api/services/GeneExpressionAnnotationService.java

```diff
@@ -0,0 +1,135 @@
+package org.alliancegenome.curation_api.services;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+import lombok.extern.jbosslog.JBossLog;
+import org.alliancegenome.curation_api.constants.EntityFieldConstants;
+import org.alliancegenome.curation_api.constants.VocabularyConstants;
+import org.alliancegenome.curation_api.dao.ExpressionAnnotationDAO;
+import org.alliancegenome.curation_api.dao.GeneExpressionAnnotationDAO;
+import org.alliancegenome.curation_api.dao.PersonDAO;
+import org.alliancegenome.curation_api.enums.BackendBulkDataProvider;
+import org.alliancegenome.curation_api.exceptions.ApiErrorException;
+import org.alliancegenome.curation_api.exceptions.ObjectUpdateException;
+import org.alliancegenome.curation_api.interfaces.crud.BaseUpsertServiceInterface;
+import org.alliancegenome.curation_api.model.entities.*;
+import org.alliancegenome.curation_api.model.entities.ontology.MMOTerm;
+import org.alliancegenome.curation_api.model.ingest.dto.fms.GeneExpressionFmsDTO;
+import org.alliancegenome.curation_api.response.ObjectResponse;
+import org.alliancegenome.curation_api.response.SearchResponse;
+import org.alliancegenome.curation_api.services.base.BaseAnnotationCrudService;
+import org.alliancegenome.curation_api.services.helpers.annotations.GeneExpressionAnnotationUniqueIdHelper;
+import org.alliancegenome.curation_api.services.ontology.MmoTermService;
+import org.alliancegenome.curation_api.services.validation.dto.fms.GeneExpressionAnnotationFmsDTOValidator;
+import org.apache.commons.lang.StringUtils;
+
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RequestScoped
+@JBossLog
+public class GeneExpressionAnnotationService extends BaseAnnotationCrudService<GeneExpressionAnnotation, GeneExpressionAnnotationDAO> implements BaseUpsertServiceInterface<GeneExpressionAnnotation, GeneExpressionFmsDTO> {
+
+	@Inject	ExpressionAnnotationDAO expressionAnnotationDAO;
+
+	@Inject	GeneExpressionAnnotationDAO geneExpressionAnnotationDAO;
+
+	@Inject GeneService geneService;
+
+	@Inject MmoTermService mmoTermService;
+
+	@Inject	ReferenceService referenceService;
+
+	@Inject GeneExpressionAnnotationUniqueIdHelper geneExpressionAnnotationUniqueIdHelper;
+
+	@Inject	GeneExpressionAnnotationFmsDTOValidator geneExpressionAnnotationFmsDTOValidator;
+
+	@Inject DataProviderService dataProviderService;
+
+	@Inject VocabularyTermService vocabularyTermService;
+
+	@Inject	PersonDAO personDAO;
+
+	@Inject PersonService personService;
```

Comment:
These imports don't look like they are formatted well. There are tabs instead of spaces.

Reply:
Fixed

Reply:
There are still tabs here where there should be a single space.